vision/module/location.py

```diff
@@ -249,6 +251,171 @@ def _get_slopes(self) -> None:
         # Convert slopes to degrees
         self.slopes = np.degrees(np.arctan(self.slopes))
 
+    ## Image Processing
+
+    def _increase_brightness(self, offset: int) -> np.ndarray:
+        """
+        Increases brightness of self.img by scaling the median to 128 + an offset
+
+        Parameters
+        ----------
+        offset: int
+            The offset at which to scale the median brightness to. Range: [-128, 127]
+
+        Returns
+        -------
+        np.ndarray - brightened color image
+        """
+        # Get gray scale of color image
+        gray = cv2.cvtColor(src=self.img, code=cv2.COLOR_RGB2GRAY)
+
+        # Calculate magnitude of brightness based on median + offset
+        alpha_value = (128 + offset) / (np.median(gray))
+
+        # Brighten image
+        brightened_image = cv2.addWeighted(
+            src1=self.img, alpha=alpha_value, src2=0, beta=0, gamma=0
+        )  # brightened_image = src1*alpha + src2*beta + gamma
+
+        return brightened_image
+
+    def _filter_text_circles(self) -> np.ndarray:
+        """
+        Filters out circles that are to the left of, right of, or above text.
+
+        Returns
+        -------
+        None
+        """
+        # Detect text
+        detector = TextDetector()
+        text_boxes = detector.detect_russian_word(self.img, self.depth)
+
+        # Return if text detection fails
+        if not text_boxes:
+            return self.circles
+
+        # Convert BoundingBoxes to list of values per axis
+        vertices = np.array(
+            [vertex for bbox in text_boxes for vertex in bbox.vertices]
+        )  # unravel into list of vertices
+        axes_vals = vertices.transpose()  # axis_vals[0] = x, axis_vals[1] = y
+
+        # Find max and min values for each axis
+        max_x = np.max(axes_vals[0])
+        min_x = np.min(axes_vals[0])
+        max_y = np.max(axes_vals[1])
+        min_y = np.min(axes_vals[1])
+
+        # Create rectangle that encompasses all detected text boxes
+        # ul_bound = (min_x, min_y) # upper left corner #NOTE: not needed
+        # ur_bound = (max_x, min_y) # upper right corner #NOTE: not needed
+        ll_bound = (min_x, max_y)  # lower left corner
+        lr_bound = (max_x, max_y)  # lower right corner
+
+        # Filter out rectangles that are not directly below text
+        circle_filter = np.array(
+            [], dtype=np.uint16
+        )  # array of indices of circles out of bounds
+
+        i = 0
+        for x, y, _ in self.circles:
```

Comment:
Could use enumerate here instead of a counter variable. Due to how Python is interpreted, counter variables are extremely slow whereas for loops can be translated into precompiled c++ code.
Would be:
```
for i, circle in enumerate(self.circles):
    x, y, _ = circle
```

relevant stackoverflow: https://stackoverflow.com/questions/522563/accessing-the-index-in-for-loops